SU2_CFD/src/solver_direct_mean.cpp

```diff
@@ -10882,15 +10892,31 @@ void CEulerSolver::BC_Fluid_Interface(CGeometry *geometry, CSolver **solver_cont
 
           /*--- Set primitive variables ---*/
 
-          numerics->SetPrimitive(PrimVar_i, PrimVar_j);
+          numerics->SetPrimitive( PrimVar_i, PrimVar_j );
+          
+          if( !( config->GetKind_FluidModel() == STANDARD_AIR || config->GetKind_FluidModel() == IDEAL_GAS ) ){
+		    P_static   = PrimVar_j[nDim+1];
+		    rho_static = PrimVar_j[nDim+2];			  
+			FluidModel->SetTDState_Prho(P_static, rho_static);
+			  			  
+		    P_static   = PrimVar_i[nDim+1];
+		    rho_static = PrimVar_i[nDim+2];
+			FluidModel->SetTDState_Prho(P_static, rho_static);
+            node[iPoint]->SetSecondaryVar_Compressible(FluidModel);
+
+            Secondary_i[0] = FluidModel->GetdPdrho_e();
+            Secondary_i[1] = FluidModel->GetdPde_rho();              
+
+            numerics->SetSecondary(Secondary_i, Secondary_j);
+          }
 
```

Comment:
Giulio, I am not sure about this maybe I am missing something. I see here that the fluid model is instantiated twice but used once. How are the Secondary_j calculated?  Why not using for the iNode simply the GetSecondary method and then calculating for j the secondary variables using the FluidModel ? Also, but I am not sure, I guess the if statement can be omitted.

What about something like this:

         	```
           Secondary_i = node[iPoint]->GetSecondary();

          	P_static   = PrimVar_j[nDim+1];
          	rho_static = PrimVar_j[nDim+2];
          	FluidModel->SetTDState_Prho(P_static, rho_static);

          	Secondary_j[0] = FluidModel->GetdPdrho_e();
          	Secondary_j[1] = FluidModel->GetdPde_rho();


          /*--- Set primitive variables ---*/
          numerics->SetPrimitive(PrimVar_i, PrimVar_j);

          /*--- Set secondary variables ---*/
          numerics->SetSecondary(Secondary_i, Secondary_j);


```